src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs

```diff
@@ -105,7 +105,8 @@ public class CookieAuthenticationEvents
     private static bool IsAjaxRequest(HttpRequest request)
     {
         return string.Equals(request.Query[HeaderNames.XRequestedWith], "XMLHttpRequest", StringComparison.Ordinal) ||
-            string.Equals(request.Headers.XRequestedWith, "XMLHttpRequest", StringComparison.Ordinal);
+            string.Equals(request.Headers.XRequestedWith, "XMLHttpRequest", StringComparison.Ordinal) ||
+            request.Headers.Accept.ToString().Contains("application/json");
```

Comment:
This seems overly broad.

Reply:
For a request against https://example.com/foo.json
Firefox sends:

    Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8

Chromium sends:

    Accept:  text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9

Edge sends:

    Accept:  text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9

So none of them sends "application/json" so this shouldn't trigger any false positives. If considered overly broad we can instead do a strict equals comparison instead of a string contains comparison if we would like to avoid to trigger it on requests such as:

    Accept: application/json,*/*